wkcuber/tile_cubing.py

```diff
@@ -1,60 +1,233 @@
+            number_of_digits = (
+                len(occurrence) - 2 - coord_ids_with_replacement_info[coord][1]
+            )
+            if number_of_digits > 1:
+                format_str = "0" + str(number_of_digits) + "d"
+            else:
+                format_str = "d"
+            pattern = pattern.replace(
+                occurrence,
+                format(coord_ids_with_replacement_info[coord][0], format_str),
+                1,
+            )
+    return pattern
+
+
+def replace_coordinates_with_glob_regex(pattern: str, coord_ids: Dict[str, int]) -> str:
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        coord = occurrence[1]
+        if coord in coord_ids:
+            number_of_digits = len(occurrence) - 2 - coord_ids[coord]
+            pattern = pattern.replace(occurrence, "[0-9]" * number_of_digits, 1)
+    return pattern
 
-    all_source_files.sort()
-    return all_source_files
 
+def get_digit_numbers_for_dimension(pattern):
+    x_number = 0
+    y_number = 0
+    z_number = 0
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        if occurrence[1] == "x":
+            x_number = max(x_number, len(occurrence) - 2)
+        if occurrence[1] == "y":
+            y_number = max(y_number, len(occurrence) - 2)
+        if occurrence[1] == "z":
+            z_number = max(z_number, len(occurrence) - 2)
+    return x_number, y_number, z_number
+
+
+def detect_interval_for_dimensions(
+    file_path_pattern: str,
+    x_decimal_length: int,
+    y_decimal_length: int,
+    z_decimal_length: int,
+) -> int:
+    x_min = None
+    y_min = None
+    z_min = None
+    x_max = 0
+    y_max = 0
+    z_max = 0
+    arbitrary_file = None
+    file_count = 0
+    for x in range(x_decimal_length):
+        for y in range(y_decimal_length):
+            for z in range(z_decimal_length):
+                specific_pattern = replace_coordinates_with_glob_regex(
+                    file_path_pattern, {"z": z, "y": y, "x": x}
+                )
+                found_files = glob(specific_pattern)
+                file_count += len(found_files)
+                for file in found_files:
+                    arbitrary_file = file
+                    occurrences = re.findall("({x+}|{y+}|{z+})", file_path_pattern)
+                    index_offset_caused_by_brackets_and_specific_length = 0
+                    for occurrence in occurrences:
+                        # update the offset since the pattern and the file path have different length
+                        occurrence_begin_index = (
+                            file_path_pattern.index(occurrence)
+                            - index_offset_caused_by_brackets_and_specific_length
+                        )
+                        index_offset_caused_by_brackets_and_specific_length += 2
+                        if occurrence[1] == "x":
```

Comment:
It should be possible to unify the three branches of this if-statement, since they are almost identical (except for variable names). If you pack x_min, y_min, z_min etc. into one dict `min_values = { "x": ..., "y": .... }` you can set the appropriate values dynamically.

Overall, this method seems quite long and I didn't grasp it fully yet. Might be easier to understand after refactoring the if-statement together with my other recommendation to reduce padding-tolerance.
We can also discuss this in person, if you want.

Reply:
Nice Idea :+1: